components/google_drive/actions/download-file/download-file.js

```diff
@@ -0,0 +1,84 @@
+const googleDrive = require("../../google_drive.app");
+const fs = require("fs");
+const stream = require("stream");
+const { promisify } = require("util");
+const { GOOGLE_DRIVE_MIME_TYPE_PREFIX } = require("../../constants");
+
+module.exports = {
+  key: "google_drive-download-file",
+  name: "Download File",
+  description: "Download a file",
+  version: "0.0.8",
+  type: "action",
+  props: {
+    googleDrive,
+    drive: {
+      propDefinition: [
+        googleDrive,
+        "watchedDrive",
+      ],
+      description: "The drive containing the file to download.",
+      optional: true,
+      default: "",
+    },
+    fileId: {
+      propDefinition: [
+        googleDrive,
+        "fileId",
+        (c) => ({
+          drive: c.drive,
+        }),
+      ],
+      description: "The file to download.",
+    },
+    filePath: {
+      type: "string",
+      label: "Destination File Path",
+      description:
+        "The destination path for the file in /tmp (e.g., `/tmp/myFile.csv`).",
+    },
+  },
+  async run() {
+    // Get file metadata to get mime type
+    // Use files.export for google file types, files.get for other file types
+    const drive = this.googleDrive.drive();
+    // Get file mimeType
+    const fileMetadata = await this.googleDrive.getFile(this.fileId, {
+      fields: "mimeType",
+    });
+    const mimeType = fileMetadata.mimeType;
+    // Download
+    let file;
+    if (mimeType.includes(GOOGLE_DRIVE_MIME_TYPE_PREFIX)) {
+      // Google MIME type
+      // See https://developers.google.com/drive/api/v3/mime-types for a list of
+      // Google MIME types.
+      file = (
+        await drive.files.export(
+          {
+            fileId: this.fileId,
+            mimeType: "application/pdf",
```

Comment:
Is `application/pdf` intentional?

Reply:
Yes, but other functionality might be preferable. Besides Apps Scripts Google Workspace Files that have a [Google Drive specific MIME type](https://developers.google.com/drive/api/v3/mime-types) can be be converted to `application/pdf` when using [export](https://developers.google.com/drive/api/v3/reference/files/export). I should add a comment about that.

Reply:
Maybe we could add a `File Type` prop to choose the type of the downloaded file, and the options can be determined by the type of the selected file. The [About resource](https://developers.google.com/drive/api/v3/reference/about#resource) in the Google Drive API contains an `exportFormats` object, which is "a map of source MIME type to possible targets for all supported exports".

Reply:
@dannyroosevelt can you sync with Jacob on this one? We should probably think through the ideal experience for this action / see what Zapier does.

Reply:
Zapier doesn't seem to have a **Download File** action.
If "Show advanced settings" is enabled, Integromat's **Download File** Action shows a prop to choose a file conversion format for each Google Workspace file type:
<img width="300" alt="downloadprops3" src="https://user-images.githubusercontent.com/19861096/134579677-6a5c75ea-ad65-4230-bb07-c6f630a05673.png">

Reply:
Added `async options` to the [`mimeType` prop in **Download File**](https://github.com/PipedreamHQ/pipedream/blob/f461bfb3793a727c705007472dbc9927e0a6579c/components/google_drive/actions/download-file/download-file.mjs#L64) which returns the following:
- If the value of the `fileId` prop is the ID of a Google Workspace document, return the [valid conversion format MIME types](https://developers.google.com/drive/api/v3/ref-export-formats) for that file type
- If the value of the `fileId` prop is the ID of a non-Google Doc file, return an empty array (because this prop is not used)
- Otherwise, return all [Google Workspace document export MIME types](https://developers.google.com/drive/api/v3/ref-export-formats)

The `mimeType` prop is optional, but an error is throw in the `run` method when the file being downloaded is a [Google Workspace document](https://developers.google.com/drive/api/v3/ref-export-formats) and `mimeType` is blank.